File: R/DataPrepration/Data_preprocessing.R
# Import Data set 
df = read.csv("C:/Users/snapp/Data-Projects/Datasets/Data_Preprocessing1.csv")

# Missing Value 
df$Age = ifelse(is.na(df$Age),
                ave(df$Age, FUN = function(x) mean(x,na.rm = TRUE)),
                df$Age)

df$Salary = ifelse(is.na(df$Salary),
                ave(df$Salary, FUN = function(x) mean(x,na.rm = TRUE)),
                df$Salary)


# Encode Categorical Data 
df$Country = factor(df$Country,
                    levels = c('France','Spain','Germany'),
                    labels = c(1,2,3))

df$Purchased = factor(df$Purchased,
                    levels = c('No','Yes'),
                    labels = c(0,1))

# Split Dataset into train test split 
# Install a library: install.packages('caTools')
library(caTools)
# Set random state for seeing same data 
set.seed(123)
# See help with F1
split = sample.split(df$Purchased, SplitRatio = 0.8)
train_set = subset(df, split == TRUE)
test_set = subset(df, split == FALSE)

# Feature Scaling 
# Need Numeric Columns 

train_set[, 2:3] = scale(train_set[, 2:3]) 
test_set[, 2:3] = scale(test_set[, 2:3])

File: Python/DataPrepration/Data_preprocessing.py
# Import Liabraries 
import numpy as np 
import matplotlib.pyplot as plt
import pandas as pd 
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, MinMaxScaler

# Import Datasets 
df = pd.read_csv(r"C:\Users\snapp\Data-Projects\Datasets\Data_Preprocessing1.csv")
# Independent Variables 
X = df.iloc[:,:-1]
# X = df.iloc[:,:-1].values
# Dependent Variables 
y = df.iloc[:,3].values 

# Missing Data 
imputer = SimpleImputer(missing_values=np.nan, strategy='mean')
imputer = imputer.fit(X.iloc[:,1:])
X.iloc[:,1:] = imputer.transform(X.iloc[:,1:])

# Encoding Categorical Data with two values
labelencoder_y = LabelEncoder()
y = labelencoder_y.fit_transform(y)

# Encoding Categorical Data with more than two values
onehotencoder = OneHotEncoder(handle_unknown='ignore')
enc_df = pd.DataFrame(onehotencoder.fit_transform(X[['Country']]).toarray())
X = X[['Age','Salary']].join(enc_df)
X = X.iloc[:,:].values

# Split Data into Train and Test 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)

# Feature Scaling: Standardisation(-1,+1) - Normalization(0,1) 
# Is it necessary to scale dummy variables? It depends

sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)

# mm_X = MinMaxScaler()
# X_train = mm_X.fit_transform(X_train)
# X_test = mm_X.transform(X_test)
